### ISSUE TITLE
Hangfire.AspNetCore package doesn't support a target of 3.1

### ISSUE THREAD

**Issue by shanerogers** [HangfireIO/Hangfire#1619]
Hey. It seems that I am unable to get a reference to the AddHangfireServer() & .AddHangfire() extension methods for the IServiceCollection abstraction in a .net core app targetting 'netcoreapp3.1'.

Taking a look at the project for these in your repo. I see you only target the following: **net451;net461;netstandard1.3;netstandard2.0;netcoreapp3.0**

Do you plan on updating this package?

**Comment by semuserable** [comment]
Also interested in this. Btw, is it hard to do at all? Can help if there are some directions provided.

**Comment by shanerogers** [comment]
@semuserable so I found a workaround. Especially seeing I’ve not heard anything in a possible update. I could do a PR ;) - maybe you could. 

The workaround is the call the hang fire methods in a separate project in the solution. The project I created is targeting standard2.0

**Comment by semuserable** [comment]
@shanerogers yeah, I was thinking just about that, but still it's a separate project which leads to an additional infrastructure work for it. It would be nice to have it in the same service which targets `netcoreapp3.1`.

Until then, yes, totally viable solution :)

**Comment by shanerogers** [comment]
Indeed. It worked out for me as I had a number of other lib in use and the start up was getting too big for my liking.

**Comment by odinserj** [comment]
Guys, what is the exact problem you have with Hangfire.AspNetCore? Despite it doesn't reference `netcoreapp3.1` target, it works there. `netstandard3.0` was added to support additional features of ASP.NET Core 3.0 as far as I remember, and nothing else.

Maybe `using Hangfire;` directive is missing? Is `HangfireServiceCollectionExtensions` class name accessible?

**Comment by shanerogers** [comment]
Hey @odinserj, of course, I tried with the using statement. ;)

I get complete access to the required methods that extend the IServiceCollection abstraction. But one in a 3.1 targeted component the methods are not there. I'm not exactly sure why you need to target these anyway? Ideally, you just need to target a .net standard.

**Comment by semuserable** [comment]
@odinserj feeling so dumb right now.. It seems I missed something the first time I tried. Yes, it's working just fine.

@shanerogers my ASP.NET Core targets netcoreapp3.1 and I just added `Hangfire` package. Now, it works. I dunno want I missed the first time (night-late coding probably).

**Comment by shanerogers** [comment]
Okay. I will take a look again today if I have time. @semuserable - Can you do me a favour please. Can you send me a copy of your .net core 3.1 app *.csprog file? You can remove the references to other projects if you don't want to share too much ;0

**Comment by semuserable** [comment]
@shanerogers you can check https://github.com/semuserable/hangfire-basic-template

**Comment by rbrugnollo** [comment]
@semuserable I saw that you used a reference to Hangfire and that works for me as well. But shouldn't we be using Hangfire.Core reference for a 3.1 project? This one doesn't work for me.

**Comment by semuserable** [comment]
@rbrugnollo `Hangfire` package has dependencies on `Hangfire.Core`, `Hangfire.SqlServer`, `Hangfire.AspNetCore`. So, by installing `Hangfire` you will have `Hangfire.Core`.

Btw, why do you think that it should be just `Hangfire.Core`? Also, it's not clear what do you mean by "doesn't work for me". How are you trying to run it, which context? If you want to use it via `ASP.NET Core`, you should have `Hangfire.AspNetCore` dependency.
